LdapForNet/RequestHandlers/CompareRequestHandler.cs

```diff
@@ -19,15 +18,15 @@ protected override int SendRequest(SafeHandle handle, DirectoryRequest request,
                     throw new LdapException(new LdapExceptionData("Wrong assertion"));
                 }
 
-                var value = compareRequest.Assertion.GetRawValues().Single();
+                var value = compareRequest.Assertion.GetRawValues()[0];
```

Comment:
why?

Reply:
Accessing the correct index here is better than using Single (which needs to instantiate an Enumerable), and also GetRawValues returns an IList to make the access easier and faster